### DISCUSSION TITLE
How to get/access globally visible model elements via their name?

### DISCUSSION THREAD

**Discussion by goto40** [eclipse-langium/langium#621] [Q&A] [answered]
How to get/access globally visible model elements via their name? 

Until now I did this with the help of the `IndexManager`. 

I have `AstNodeDescription` elements from
`indexmanger.allElements`. I found the description based on the name. The `description.node` is undefined, as expected (specified in the documentation of `AstNodeDescription`: the found description is not local). What is the preferred way to get the `AstNode`? Is there a function for that? or do I have to manually get the document (via `LangiumDocuments`) and find the node in `document.parseResult.value`?

**Comment by msujew** [comment, marked as answer]
Hey @goto40,

once you have the appropriate `AstNodeDescription` element it's relatively simple to find the actual node. We do the following in the `DefaultLinker` during the linking process:

https://github.com/langium/langium/blob/34b37014727694b7f09e969ecc13f74cf735f926/packages/langium/src/references/linker.ts#L205-L211

**Reply by goto40** [reply]
Perfect! Thank you, @msujew.
